CRM/Mailing/Event/BAO/Reply.php

```diff
@@ -220,47 +214,32 @@ private static function autoRespond(&$mailing, $queue_id, $replyto) {
     $component->id = $mailing->reply_id;
     $component->find(TRUE);
 
-    $domain = CRM_Core_BAO_Domain::getDomain();
-    list($domainEmailName, $domainEmailAddress) = CRM_Core_BAO_Domain::getNameAndEmail();
+    [$domainEmailName, $domainEmailAddress] = CRM_Core_BAO_Domain::getNameAndEmail();
 
     $params = [
-      'subject' => $component->subject,
       'toEmail' => $to,
       'from' => "\"{$domainEmailName}\" <{$domainEmailAddress}>",
       'replyTo' => CRM_Core_BAO_Domain::getNoReplyEmailAddress(),
       'returnPath' => CRM_Core_BAO_Domain::getNoReplyEmailAddress(),
     ];
 
-    // TODO: do we need reply tokens?
-    $html = $component->body_html;
-    if ($component->body_text) {
-      $text = $component->body_text;
-    }
-    else {
-      $text = CRM_Utils_String::htmlToText($component->body_html);
-    }
-
-    $bao = new CRM_Mailing_BAO_Mailing();
-    $bao->body_text = $text;
-    $bao->body_html = $html;
-    $tokens = $bao->getTokens();
-
-    if ($eq->format == 'HTML' || $eq->format == 'Both') {
-      $html = CRM_Utils_Token::replaceDomainTokens($html, $domain, TRUE, $tokens['html']);
-      $html = CRM_Utils_Token::replaceMailingTokens($html, $mailing, NULL, $tokens['html']);
-    }
-    if (!$html || $eq->format == 'Text' || $eq->format == 'Both') {
-      $text = CRM_Utils_Token::replaceDomainTokens($text, $domain, FALSE, $tokens['text']);
-      $text = CRM_Utils_Token::replaceMailingTokens($text, $mailing, NULL, $tokens['text']);
-    }
-    $params['html'] = $html;
-    $params['text'] = $text;
+    $html = CRM_Utils_Token::replaceMailingTokens($component->body_html, $mailing, NULL, $component->body_html);
+    $text = CRM_Utils_Token::replaceMailingTokens($component->body_text, $mailing, NULL, $component->body_text);
 
     CRM_Mailing_BAO_Mailing::addMessageIdHeader($params, 'a', $eq->job_id, $queue_id, $eq->hash);
     if (CRM_Core_BAO_MailSettings::includeMessageId()) {
       $params['messageId'] = $params['Message-ID'];
     }
-    CRM_Utils_Mail::send($params);
+    \CRM_Core_BAO_MessageTemplate::sendTemplate([
+      'envelope' => $params,
```

Comment:
@eileenmcnaughton I'm not seeing how envelope key is being used I would have thought that this should be

``` 
\CRM_Core_BAO_MessageTemplate::sentTemplate(array_merge($params, [
   'tokenContext => ...
 ]));
```

Reply:
yeah - that function is horrible - really confusing. We really need a sane api wrapper for it

Reply:
Well writing the unit test helped me prove that thing works so yeh bloody confusing that is for sure